### PR TITLE
Avoid using unspecified keyword args where possible

### PR DESCRIPTION
**Pull request by murgatroid99**
I believe that this change is necessary to make room for the compression API, and any other APIs we want to add in the future.

### CODE REVIEW

**Comment by jtattermusch** [comment]
interop tests are failing:
```
Using polling engine: poll
/var/local/git/grpc/src/ruby/lib/grpc/generic/service.rb:171:in `initialize': unknown keyword: grpc.ssl_target_name_override (ArgumentError)
```

and ruby performance tests are timing out (possibly related).

File: src/ruby/lib/grpc/generic/active_call.rb [hunk 1]
```diff
@@ -44,7 +44,7 @@ def check_status
         # raise BadStatus, propagating the metadata if present.
         md = status.metadata
         with_sym_keys = Hash[md.each_pair.collect { |x, y| [x.to_sym, y] }]
```
Comment (jtattermusch) [inline src/ruby/lib/grpc/generic/active_call.rb]: btw, the metadata value should be a multi-map (and it looks like a ruby Hash isn't)

Comment (jtattermusch) [inline src/ruby/lib/grpc/generic/active_call.rb]: Let's consider addressing this later (fixing in this PR would make things too complicated).

**Comment by jtattermusch** [comment]
The change looks good, will take another quick look after the issues are fixed.

Thanks for taking the time to remove the very very wrong concept of metadata keys as keyword args.

File: src/ruby/lib/grpc/generic/active_call.rb (hunk 1 above)
Comment (murgatroid99) [inline src/ruby/lib/grpc/generic/active_call.rb]: A value in the hash can be a list of strings, which is handled appropriately in the core.

**Comment by murgatroid99** [comment]
The interop tests are green, and so is everything else except for a couple of *san failures in core

**Comment by jtattermusch** [comment]
LGTM.

**Pull request merged**